src/adr/0034-distributed-listeners-configuration.md

```diff
@@ -0,0 +1,179 @@
+perform additional mailbox listener registry with this WebAdmin endpoint. JMAP SpamAssassin, quota mailing tests are 
+concerned.
+
+We will also expose an endpoint listing the groups currently in use, and for each group the associated configuration, if 
+any. This will query the **configured mailbox listener aggregate**.
+
+We will introduce a health check to actually ensure that RabbitMQ resources match the configured listeners, and propose
+a WebAdmin endpoint to add/remove bindings/queue in a similar fashion of what had been proposed in 
+[26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md). This 
+can happen if the James server performing the listener registration fails to create the group/queue. 
+
+## Consequences
+
+All products other than "Distributed James" are unchanged.
+
+All the currently configured additional listeners will need to be registered.
+
+The definition of mailbox listeners is thus centralized and we are not exposed to an heterogeneous configuration 
+incident.
+
+Mailbox listeners no longer required by guice will still need to be instanciable (even with empty content). They will 
+be considered as additional listener thus requiring explicit admin unconfiguration, which will be mentioned in the 
+related upgrade instructions. Read notes about [rolling upgrade scenarii](#rolling-upgrade-scenari).
+
+[Deploying a new custom listener](#deploying-a-new-custom-listener) also describes how to deploy new custom listeners.
+
+## Notes
+
+## Broadcast of topology changes
+
+A broadcast can be attempted to propagate eventBus topology changes:
+
+ - Each James server registers an exclusive queue to an "eventBus topology change" exchange.
+ - Upon modification of the actual topology an "add" or "remove" event is emitted.
+ - Each running James reacts to these events by instantiating the corresponding listener and starting consuming the 
+ associated queue, or stops consuming the associated queue.
+ 
+If a listener is added but is not in the classpath, an ERROR log is emitted. This can happen during a rolling upgrade,
+which defines a new guice binding for a new mailbox listener. Events will still be emitted (and consumed by other James
+servers) however a local James upgrade will be required to effectively be able to start processing these events. The 
+binding will not need to be redefined.
+
+Propagating changes will thus no longer need server reboot.
+
+This won't be contributed in the short run as it requires significant development effort.
+
+### Rolling upgrade scenari
+
+During a rolling upgrade, the james version is heterogeneous across the cluster, and so might be the mailbox listeners
+required at the Guice level.
+
+**case 1**: James Server version 1 does not require listener A, James server version 2 requires listener A.
+
+Since listener A is registered, James server version 1 can not be rebooted prior being upgraded. (As listener A cannot be instanciated)
```

Comment:
```suggestion
Since listener A is registered, James server version 1 cannot be rebooted without being upgraded first. (As listener A cannot be instantiated)
```